examples/ethernet/ethernet/main/ethernet_main.c

```diff
@@ -32,92 +32,45 @@
 #include "tcpip_adapter.h"
 #include "nvs_flash.h"
 #include "driver/gpio.h"
-#include "tlk110_phy.h"
 
-static const char *TAG = "eth_demo";
+extern void phy_device_check_phy_init(void);
```

Comment:
There are a couple of considerations:
* We try to use code-as-configuration where possible instead of sdkconfig. This often leads to more readable code (if each PHY symbol has a different name, it's obvious what the intention of the phy configuration selection is from reading the calling code in ethernet_main. Whereas using the same name to configure PHYs means a reader has to also look at the component.mk file in order to see the full picture of what's happening, and how the three parts - sdkconfig, build configuration, ethernet_main - join together.)
* The second reason is that for Arduino and some other environments (like Micropython/NodeMCU) the maintainers build a single set of libraries and then link to those. For these environments to support multiple PHYs, they have to be able to access them all simultaneously.
Thanks for being patient with all my requests. :)